.github/workflows/ddd-validation.yml

```
@@ -0,0 +1,222 @@
+  validate-framework:
+    runs-on: ubuntu-latest
+    name: DDD Framework Compliance Check
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Need history for git pattern validation
+    
+    - name: Set up Go
+      uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)